arcgis-ios-sdk-samples/Utility network/Display content of utility network container/DisplayContentUtilityNetworkContainerViewController.swift

```diff
@@ -0,0 +1,366 @@
+                    self.graphicsOverlay.graphics.add(AGSGraphic(geometry: association.geometry, symbol: symbol))
+                }
+                // Enable the bar button item to exit the container view.
+                self.exitBarButtonItem.isEnabled = true
+            }
+        }
+    }
+    
+    // MARK: - AGSGeoViewTouchDelegate
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Turn off user interaction to avoid straying away from the container view.
+        self.mapView.isUserInteractionEnabled = false
+        // Identify the top most feature that corresponds to the tapped point.
+        mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 5, returnPopupsOnly: false) { [weak self] (layerResults, error) in
+            guard let self = self else { return }
+            if let layerResults = layerResults {
+                self.identifyContainerFeature(layerResults: layerResults) { containerFeature in
+                    self.addElementAssociations(for: containerFeature)
+                }
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+    }
+    
+    // MARK: - UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Load the utility network.
+        loadUtilityNetwork()
+        // Add a graphics overlay.
+        mapView.graphicsOverlays.add(graphicsOverlay)
+        // Get the legends from the feature service.
+        fetchLegendInfo()
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["DisplayContentUtilityNetworkContainerViewController", "DisplayContentUtilityNetworkTableViewController"]
+    }
+    
+    // MARK: - Navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "DisplayContentLegendSegue",
+            let controller = segue.destination as? UITableViewController {
+                controller.presentationController?.delegate = self
+                controller.tableView.dataSource = symbolsDataSource
+            }
+    }
+    
+    // MARK: - UIAdaptivePresentationControllerDelegate
+    func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
+        return .none
+    }
+}
+
+// MARK: - AGSAuthenticationManagerDelegate
+extension DisplayContentUtilityNetworkContainerViewController: AGSAuthenticationManagerDelegate {
+    func authenticationManager(_ authenticationManager: AGSAuthenticationManager, didReceive challenge: AGSAuthenticationChallenge) {
+        // NOTE: Never hardcode login information in a production application. This is done solely for the sake of the sample.
+        let credentials = AGSCredential(user: "editor01", password: "S7#i2LWmYH75")
```

Comment:
Good idea, I've changed the credentials to use viewer01. In this case, the map is loaded from the web map which requires authentication because it's hosted on the sample server 7 portal. The authentication from loading the web map doesn't transfer when loading the UN. Generally, we'll only use the authentication manager in situations like this where we'll be prompted multiple separate times for credentials.

Reply:
Nice! Thanks.